gcloud/_helpers.py

```diff
@@ -238,3 +241,35 @@ def __init__(self, project=None, implicit=False):
 
 
 _DEFAULTS = _DefaultsContainer(implicit=True)
+
+
+class _ClientProxy(object):
```

Comment:
There is no "welding":  the proxies are thin wrappers, used only if the developer chooses to use the client pattern.  As with other APIs[1], this PR adds a "convenience" layer on top of a more explicit API.

Consider how SQLAlchemy[1] layers the ORM over the Core, and, within the ORM itself, how it allows separation of class design and data mapping.

Moving knowledge of the client down into the existing topic / subscription instances collapses the layers.

[1] http://www.sqlalchemy.org/features.html